src/dialects/oracle/query-generator.js

```diff
@@ -0,0 +1,1443 @@
+const JSON_FUNCTION_REGEX = /^\s*((?:[a-z]+_){0,2}jsonb?(?:_[a-z]+){0,2})\([^)]*\)/i;
+const JSON_OPERATOR_REGEX = /^\s*(->>?|@>|<@|\?[|&]?|\|{2}|#-)/i;
+const TOKEN_CAPTURE_REGEX = /^\s*((?:([`"'])(?:(?!\2).|\2{2})*\2)|[\w\d\s]+|[().,;+-])/i;
+
+class OracleQueryGenerator extends AbstractQueryGenerator {
+  constructor(options) {
+    super(options);
+  }
+
+  /**
+   * Returns the value as it is stored in the Oracle DB
+   *
+   * @param {string} value
+   */
+  getCatalogName(value) {
+    if (value) {
+      if (this.options.quoteIdentifiers === false) {
+        const quotedValue = this.quoteIdentifier(value);
+        if (quotedValue === value) {
+          value = value.toUpperCase();
+        }
+      }
+    }
+    return value;
+  }
+
+  /**
+   * Returns the tableName and schemaName as it is stored the Oracle DB
+   *
+   * @param {object|string} table
+   */
+  getSchemaNameAndTableName(table) {
+    const tableName = this.getCatalogName(table.tableName || table);
+    const schemaName = this.getCatalogName(table.schema);
+    return [tableName, schemaName];
+  }
+
+  createSchema(schema) {
+    const quotedSchema = this.quoteTable(schema);
+    const schemaName = this.getCatalogName(schema);
+    return [
+      'DECLARE',
+      '  V_COUNT INTEGER;',
+      '  V_CURSOR_NAME INTEGER;',
+      '  V_RET INTEGER;',
+      'BEGIN',
+      '  SELECT COUNT(1) INTO V_COUNT FROM ALL_USERS WHERE USERNAME = ',
+      wrapSingleQuote(schemaName),
+      ';',
+      '  IF V_COUNT = 0 THEN',
+      '    EXECUTE IMMEDIATE ',
+      wrapSingleQuote(`CREATE USER ${ quotedSchema } IDENTIFIED BY 12345 DEFAULT TABLESPACE USERS`),
+      ';',
+      '    EXECUTE IMMEDIATE ',
+      wrapSingleQuote(`GRANT "CONNECT" TO ${quotedSchema}`),
+      ';',
+      '    EXECUTE IMMEDIATE ',
+      wrapSingleQuote(`GRANT create table TO ${quotedSchema}`),
+      ';',
+      '    EXECUTE IMMEDIATE ',
+      wrapSingleQuote(`GRANT create view TO ${quotedSchema}`),
+      ';',
+      '    EXECUTE IMMEDIATE ',
+      wrapSingleQuote(`GRANT create any trigger TO ${quotedSchema}`),
+      ';',
+      '    EXECUTE IMMEDIATE ',
+      wrapSingleQuote(`GRANT create any procedure TO ${quotedSchema}`),
+      ';',
+      '    EXECUTE IMMEDIATE ',
+      wrapSingleQuote(`GRANT create sequence TO ${quotedSchema}`),
+      ';',
+      '    EXECUTE IMMEDIATE ',
+      wrapSingleQuote(`GRANT create synonym TO ${quotedSchema}`),
+      ';',
+      '    EXECUTE IMMEDIATE ',
+      wrapSingleQuote(`ALTER USER ${quotedSchema} QUOTA UNLIMITED ON USERS`),
+      ';',
+      '  END IF;',
+      'END;'
+    ].join(' ');
+  }
+
+  showSchemasQuery() {
+    return 'SELECT USERNAME AS "schema_name" FROM ALL_USERS WHERE COMMON = (\'NO\') AND USERNAME != user';
+  }
+
+  dropSchema(schema) {
```

Comment:
Save a round trip and just execute the DROP USER.  Capture and ignore the 'user doesn't exist' error.  
Similar to the way you do in `dropTableQuery()`

https://stackoverflow.com/a/1801453/4799035

Reply:
Done